File: fe/lib/models/webscraper/JobResponseDTO.dart
import 'package:json_annotation/json_annotation.dart';

part 'JobResponseDTO.g.dart';

@JsonSerializable()
class JobResponseDTO {
  String? title;
  String? subTitle;
  String? link;
  String? location;
  String? salary;
  String? imgLink;

  JobResponseDTO({
    required this.title,
    required this.subTitle,
    this.link,
    this.location,
    this.salary,
    this.imgLink
  });

  factory JobResponseDTO.fromJson(Map<String, dynamic> json) => _$JobResponseDTOFromJson(json);
  Map<String, dynamic> toJson() => _$JobResponseDTOToJson(this);
}

File: ai-cv-generator-backend/src/main/java/com/revolvingSolutions/aicvgeneratorbackend/model/webscrapper/JobResponseDTO.java
package com.revolvingSolutions.aicvgeneratorbackend.model.webscrapper;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class JobResponseDTO {
    public String title;
    public String subTitle;
    public String link;
    public String location;
    public String imgLink;
    public String salary;
}
